### ISSUE TITLE
Adding filter plugin API

### ISSUE THREAD

**Issue by frsyuki** [embulk/embulk#26]
We want to apply conversion or filtering to records when:

* hash: applying a hash function to sensitive columns (e.g. account_id, email)
* grep: skipping a record if it matches certain condition (e.g. skip access logs if they're internal callback requests)
* convert column type: convert created_at:string column to timestamp
* skip column: convert comment field to nil if it's too long
* remove column: remove certain columns (e.g. remove "password" column)
* join: joining data with other data stores. (e.g. join email, age, and other user information to user_id)

**Comment by frsyuki** [comment]
Probably, configuration file is like this:

```
in:
  type: ...
filters:
  - type: filter_a
    key: value
  - type: filter_b
    key: value
out:
  type: ...
```

**Comment by frsyuki** [comment]
Implemented by #49
